UndertaleModLib/Compiler/Parser.cs

```diff
@@ -439,8 +446,17 @@ public static Statement ParseTokens(CompileContext context, List<Lexer.Token> to
                     context.LocalVars["arguments"] = "arguments";
                 context.GlobalVars.Clear();
                 context.Enums.Clear();
+                context.FunctionsToObliterate.Clear();
+                uuidCounter = 0;
                 hasError = false;
 
+                usableStructNames.Clear();
+                foreach (UndertaleCode child in context.OriginalCode.ChildEntries)
+                {
+                    if (child.Name.Content.StartsWith("gml_Script____struct___"))
+                        usableStructNames.Enqueue(child.Name.Content["gml_Script_".Length..]);
```

Comment:
Is this entirely exhaustive? Depending on case, structs may not start with this, i.e. if embedded inside an anonymous function (`anon` part comes first in those cases). Although, I'm unsure if anonymous functions are supported by this compiler right now anyway.

Reply:
> Depending on case, structs may not start with this, i.e. if embedded inside an anonymous function (anon part comes first in those cases).

At least from my tests in GMS2022.9, that does not seem to be the case (although it might be different in earlier or later versions).
![image](https://github.com/krzys-h/UndertaleModTool/assets/68464103/0602c954-f4de-4030-b490-8c7e1b840edf)


> Although, I'm unsure if anonymous functions are supported by this compiler right now anyway.

Sort of, but they don't work that well.
![image](https://github.com/krzys-h/UndertaleModTool/assets/68464103/55ba817a-0c66-4375-acb8-2a8b7031c737)
![image](https://github.com/krzys-h/UndertaleModTool/assets/68464103/9c7ef9a5-bc0a-465b-9039-43f3a104a13c)

Reply:
AFAIK there's an open issue somewhere about anonymous functions, to my understanding their status essentially boils down to "unsupported/not working"

Reply:
OK so I further investigated this, and I found out that the code entries I saw were just anonymous functions declared inside of structs, which aren't themselves structs, so this check is fine!